### PR TITLE
refactor: use new-style formatting for named placeholders

### PR DESCRIPTION
**Pull request by nejch**
Basically just https://pyformat.info/#named_placeholders with some sed. Similar to the f-strings, just getting rid of more python2 stuff. Although this one might have some performance penalty I'm not sure it's noticeable

### CODE REVIEW

File: tests/functional/api/test_merge_requests.py [hunk 1]
```diff
@@ -27,9 +27,11 @@ def test_merge_requests(project):
             "commit_message": "New commit in new branch",
         }
     )
-    project.mergerequests.create(
+
+    mr = project.mergerequests.create(
         {"source_branch": "branch1", "target_branch": "main", "title": "MR readme2"}
     )
+    assert mr.id is not None
```
Comment (JohnVillalovos) [inline tests/functional/api/test_merge_requests.py:34]: What's this?

File: gitlab/v4/objects/commits.py [hunk 2]
```diff
@@ -202,10 +202,10 @@ def create(
         # project_id and commit_id are in the data dict when using the CLI, but
         # they are missing when using only the API
         # See #511
-        base_path = "/projects/%(project_id)s/statuses/%(commit_id)s"
+        base_path = base_path = "/projects/{project_id}/statuses/{commit_id}"
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/commits.py:205]: This looks wrong to me, or at least a redundant `=` sign.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]
The one redundant `=` sign should be fixed. Also seems like unrelated test change.

File: tests/functional/api/test_merge_requests.py (hunk 1 above)
Comment (nejch) [reply tests/functional/api/test_merge_requests.py]: Wow thanks for that, brainfart during rebase 😁 fixed.

File: gitlab/v4/objects/commits.py (hunk 2 above)
Comment (nejch) [reply gitlab/v4/objects/commits.py]: Same as above 🤦 thanks

**Review by JohnVillalovos — APPROVED** [review]

**Pull request merged**